plugins/BEdita/Core/src/Model/Table/PropertyTypesTable.php

```diff
@@ -84,4 +102,102 @@ public function buildRules(RulesChecker $rules)
 
         return $rules;
     }
+
+    /**
+     * Invalidate object types cache after updating a property type.
+     *
+     * @return void
+     */
+    public function afterSave()
+    {
+        Cache::clear(false, ObjectTypesTable::CACHE_CONFIG);
+    }
+
+    /**
+     * Check that no properties exist linked to the property type before deleting it.
+     *
+     * @param \Cake\Event\Event $event Dispatched event.
+     * @param \Cake\Datasource\EntityInterface $entity Entity being deleted.
+     * @return void
+     * @throws \Cake\Network\Exception\ForbiddenException Throws an exception if one or more properties exist
+     *      with the property type being deleted.
+     */
+    public function beforeDelete(Event $event, EntityInterface $entity)
+    {
+        if ($this->Properties->exists([$this->Properties->getForeignKey() => $entity->get($this->Properties->getBindingKey())])) {
+            throw new ForbiddenException(__d('bedita', 'Property type with existing properties'));
+        }
+    }
+
+    /**
+     * Invalidate object types cache after deleting a property type.
+     *
+     * @return void
+     */
+    public function afterDelete()
+    {
+        Cache::clear(false, ObjectTypesTable::CACHE_CONFIG);
+    }
+
+    /**
+     * Detect most appropriate property type for a column.
+     *
+     * @param string $name Column name.
+     * @param \Cake\ORM\Table $table Table object.
+     * @return \BEdita\Core\Model\Entity\PropertyType
+     */
+    public function detect($name, Table $table)
+    {
+        /* @var \BEdita\Core\Model\Entity\PropertyType[] $propertyTypes */
+        $propertyTypes = Cache::remember(
+            'property_types',
+            function () {
+                return $this->find()
+                    ->indexBy('name')
+                    ->toArray();
+            },
+            ObjectTypesTable::CACHE_CONFIG
+        );
+
+        // Check if there is a property type whose name matches column name.
+        if (isset($propertyTypes[$name])) {
+            return $propertyTypes[$name];
+        }
+
+        // Check if there is a property type whose name matches an applicable validation rule's name.
+        $rules = array_keys($table->getValidator()->field($name)->rules());
+        foreach ($rules as $ruleName) {
+            if (isset($propertyTypes[$ruleName])) {
+                return $propertyTypes[$ruleName];
+            }
+        }
+
+        // Try to infer a generic type from column definition.
+        $type = $table->getSchema()->getColumnType($name);
+        if (isset($propertyTypes[$type])) {
+            return $propertyTypes[$type];
+        }
+        switch ($type) {
+            // Try to convert specific types to more generic ones.
+            case 'integer':
+                $type = 'number';
```

Comment:
as said before: what about an `integer` type?

Reply:
In that case we might just drop this `case` branch.